ckanext/opendata_theme/processor.py

```diff
@@ -0,0 +1,397 @@
+from abc import abstractmethod
+from collections import defaultdict, OrderedDict
+import wcag_contrast_ratio as contrast
+
+__all__ = ['custom_style_processor']
+
+from ckanext.opendata_theme.color_conrast import get_contrast
+
+
+class AbstractParser:
+    color = None
+
+    @abstractmethod
+    def get_css_from_data(self, data):
+        return
+
+    def parse_form_data(self, data):
+        value = data.get(self.get_form_name())
+        self.color = value
+
+    @abstractmethod
+    def get_class_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_form_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_title(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_default_color(self):
+        raise NotImplementedError
+
+
+class AccountHeaderBackGroundColor(AbstractParser):
+    def get_css_from_data(self, data):
+        self.parse_form_data(data)
+        if self.color:
+            return {'background': self.color.encode('utf-8')}
+
+    def get_class_name(self):
+        return '.account-masthead'
+
+    def get_form_name(self):
+        return 'custom-css-account-header-background-color'
+
+    def get_title(self):
+        return 'Account Header Background Color'
+
+    def get_default_color(self):
```

Comment:
I think it is possible to remove this method and set the default value to the color field on the class level.
the functionality will be the same, but with fewer code lines.